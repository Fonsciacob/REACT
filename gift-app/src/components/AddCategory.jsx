import React, { useState } from 'react';

export const AddCategory = ({ onNewCategory }) => {
  const [data, setData] = useState('');

  const onInputCategory = ({ target }) => {
    setData(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (data.trim().length <= 1) return;

    onNewCategory(data.trim());
    setData('');
  };

  return (
    <form onSubmit={onSubmit}>
      <input type="text" value={data} onChange={onInputCategory}></input>
    </form>
  );
};
