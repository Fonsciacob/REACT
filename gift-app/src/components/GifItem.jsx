import PropTypes from 'prop-types';

export const GifItem = ({ url }) => {
  return (
    <div className="p-2 overflow-hidden">
      <img src={url} className="" />
    </div>
  );
};

GifItem.propTypes = {
  url: PropTypes.string.isRequired,
};
