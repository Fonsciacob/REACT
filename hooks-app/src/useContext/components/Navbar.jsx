import { Link, NavLink } from 'react-router-dom';

export const Navbar = () => {
  return (
    <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
      <div className="container-fluid">
        <Link className="navbar-brand" to="/">
          Navbar
        </Link>
        <button
          class="navbar-toggler"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbarScroll"
          aria-controls="navbarScroll"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span class="navbar-toggler-icon"></span>
        </button>
        <div className="collapse navbar-collapse" id="navbarScroll">
          <ul className="navbar-nav me-auto my-2 my-lg-0 navbar-nav-scroll">
            <li className="nav-item">
              {' '}
              <NavLink
                className={({ isActive }) =>
                  `nav-link ${isActive ? 'active' : ''}`
                }
                to="/"
              >
                Home
              </NavLink>
            </li>
            <li className="nav-item">
              <NavLink
                className={({ isActive }) =>
                  `nav-link ${isActive ? 'active' : ''}`
                }
                to="/about"
              >
                About
              </NavLink>
            </li>
            <li className="nav-item">
              <NavLink
                className={({ isActive }) =>
                  `nav-link ${isActive ? 'active' : ''}`
                }
                to="/login"
              >
                login
              </NavLink>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
};
