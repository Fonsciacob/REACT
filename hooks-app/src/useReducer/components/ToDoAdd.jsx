import { useForm } from '../../hooks/useForm';

export const ToDoAdd = ({ handleNewToDo }) => {
  const { description, onInputChange, onResetForm } = useForm({
    description: '',
  });

  const onFormSubmit = () => {
    event.preventDefault();

    if (description.length <= 1) return;

    const newTodo = {
      id: new Date().getTime(),
      description,
      done: false,
    };

    handleNewToDo(newTodo);
    onResetForm();
  };

  return (
    <>
      <form
        onSubmit={onFormSubmit}
        className="mt-4 flex flex-wrap items-center justify-end"
      >
        <div className="w-full h-14 overflow-auto">
          <input
            type="text"
            className="w-full bg-slate-300 p-2 rounded focus:border-blue-500 focus:border-2 outline-none"
            name="description"
            placeholder="Add new task"
            value={description}
            onChange={onInputChange}
          />
        </div>
      </form>
    </>
  );
};
