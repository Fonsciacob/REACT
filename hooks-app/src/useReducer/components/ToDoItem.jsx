import React from 'react';

export const ToDoItem = ({ todo, onDelete, onToggleTodo }) => {
  return (
    <>
      <li className="mt-2 w-full flex justify-between items-center">
        <span
          className={`${todo.done ? 'line-through' : ''}`}
          onClick={() => onToggleTodo(todo.id)}
        >
          {todo.description}
        </span>
        <button
          onClick={() => onDelete(todo.id)}
          className="bg-red-500 px-2 py-1.5 rounded text-white"
        >
          delete
        </button>
      </li>
      <span className="w-full h-0.5 bg-slate-200 mt-2"></span>
    </>
  );
};
