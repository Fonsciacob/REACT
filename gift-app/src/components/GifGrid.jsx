import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from '../components';

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {isLoading && <h2>Loading</h2>}
      <div className="d-flex flex-wrap p-3 ">
        {images.map((images) => (
          <GifItem key={images.id} {...images} />
        ))}
      </div>
    </>
  );
};
