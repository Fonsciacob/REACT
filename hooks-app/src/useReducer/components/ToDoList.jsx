import { ToDoItem } from './ToDoItem';

export const ToDoList = ({ todos = [], onDelete, onToggleTodo }) => {
  return (
    <>
      <ul className="flex justify-around flex-wrap">
        {todos.map((todo) => (
          <ToDoItem
            key={todo.id}
            todo={todo}
            onDelete={onDelete}
            onToggleTodo={onToggleTodo}
          />
        ))}
      </ul>
    </>
  );
};
