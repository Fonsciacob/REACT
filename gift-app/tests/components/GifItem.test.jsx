import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem/>', () => {
  const url = 'https://one-punch.com/saitama.jpg';

  test('release snapshot', () => {
    const { container } = render(<GifItem url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('Show image with URL and the indicated ALT', () => {
    render(<GifItem url={url} />);
    //expect(screen.getByRole('img').src).toBe(url);
    const { src } = screen.getByRole('img');
    expect(src).toBe(url);
  });
});
