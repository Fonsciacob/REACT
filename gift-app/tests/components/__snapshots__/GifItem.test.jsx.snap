// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> release snapshot 1`] = `
<div>
  <div
    class="p-2 overflow-hidden"
  >
    <img
      class=""
      src="https://one-punch.com/saitama.jpg"
    />
  </div>
</div>
`;
